theano/gpuarray/tests/test_ctc.py

```diff
@@ -0,0 +1,188 @@
+from __future__ import (division, absolute_import, print_function)
+
+import unittest
+import numpy as np
+
+import theano
+import theano.tensor as T
+from theano.tests import unittest_tools as utt
+import theano.gpuarray
+from theano.gpuarray.ctc import (ctc_enabled, gpu_ctc, GpuConnectionistTemporalClassification)
+from theano.tensor.nnet.ctc import (ctc, ConnectionistTemporalClassification)
+from .config import (mode_with_gpu, mode_without_gpu)
+
+
+class TestCTC(unittest.TestCase):
```

Comment:
This copies a lot of the testing code from the CPU tests.  It should probably be refactored so that the CPU test can be reused with some parametrization.

Also, we don't really need to test direct usage since we expect people to use the CPU ops and rely on the optimizations to move to the GPU.